src/main/java/com/google/gwt/core/client/JavaScriptObject.java

```diff
@@ -144,7 +146,11 @@ public native String toSource() /*-{
    */
   @Override
   public final String toString() {
-    return JavaScriptObject.class.desiredAssertionStatus() ?
-        toStringVerbose(this) : toStringSimple(this);
-  }
+      if (true) {
+          return Document.Instance.toStringVerbose(this);
+      } else {
+          return JavaScriptObject.class.desiredAssertionStatus() ?
```

Comment:
Having reference to original code helps to understand differences in Mock and real implementation.
Also there is a still possibility  that I will do the same exercise for GWT 2.8.